### ISSUE TITLE
`unused_io_amount` does not fire on `match file.write() { Ok(_) }`

### ISSUE THREAD

**Issue by jyn514** [rust-lang/rust-clippy#11713]
### Summary

Clippy treats `Ok(_)` as a use of the number of bytes written.

### Lint Name

unused_io_amount

### Reproducer

I tried this code:

```rust
use std::io::Write;

fn main() {
    let mut file = std::fs::File::open("/foo").unwrap();
    match file.write(b"very large string") {
        Ok(_) => println!("yay it works"), // oops
        Err(err) => eprintln!("error {err}"),
    }
}
```

I expected to see this happen: Clippy warns that `write` is only called once and the byte size is discarded.

Instead, this happened: No warning.

### Version

```text
clippy 0.1.75 (cd674d61 2023-10-24)
rustc 1.75.0-nightly (cd674d617 2023-10-24)
binary: rustc
commit-hash: cd674d61790607dfb6faa9d754bd3adfa13aea7c
commit-date: 2023-10-24
host: aarch64-apple-darwin
release: 1.75.0-nightly
LLVM version: 17.0.3
```
